### DISCUSSION TITLE
Inject routes on startup

### DISCUSSION THREAD

**Discussion by bmsuseluda** [vercel/next.js#23308] [Ideas]
### Describe the feature you'd like to request

In our application we need to fetch the routes from a sperate url tool on startup. Therefore we  create express routes instead of using the next file-system based router.

With incremental static regeneration we are facing the problem that we don't have access to the router in getStaticProps and therefore have no information about route params.

### Describe the solution you'd like

We would like to have a more dynamic approach to create routes on startup. A callback in the next.config.js could be a solution.

### Describe alternatives you've considered

An alternative could be to add the routes in the NextServerConstructor on startup.

**Comment by sharmapukar217** [comment]
did you find any way?
